internal/Yesod/Session/Persist/Cookie/Reading.hs

```diff
@@ -0,0 +1,24 @@
+module Yesod.Session.Persist.Cookie.Reading
+  ( findSessionKey
+  )
+where
+
+import Yesod.Session.Persist.Prelude
+
+import Network.HTTP.Types.Header
+import Network.Wai
+import Web.Cookie
+
+-- | Find a session key in the request
+findSessionKey :: ByteString -> Request -> Maybe ByteString
+findSessionKey cookieNameBS =
+  one
+    . concatMap (lookupAll cookieNameBS . parseCookies)
+    . lookupAll hCookie
+    . requestHeaders
+
+one :: [a] -> Maybe a
+one = \case [x] -> Just x; _ -> Nothing
```

Comment:
Can we just use `Data.Maybe (listToMaybe)`?

Reply:
I think to be safe we should reject cases where the client presents us with more than one session cookie.